shard/src/api/api.py

```diff
@@ -73,8 +77,47 @@ async def set_response(self, user_id, guild_id, trigger, response):
 
     @fetch_guild
     async def get_playlist(self, guild):
-        voice = self.bot.cogs['Voice'].voice_managers[guild.id]
-        return voice.q.as_dict(), sc.OK_200
+        voice = self.bot.lavalink.player_manager.get(guild)
+        if voice is None:
+            return {}, sc.OK_200
+        else:
+            songs = []
+            for s in voice.queue:
+                songs.append({'title': s.title, 'author': s.author, 'duration': s.duration, 'uri': s.uri})
+            return {'songs': songs}, sc.OK_200
+
+    @fetch_guild
+    async def queue_song(self, guild, uid, song):
```

Comment:
does this correctly handle if the user is a: not in the guild and b: not in the correct voice channel

Reply:
How would the jwt stuff validate if the user isn't in the guild?
It does not handle if the user is not in the correct voice channel. Nothing was implemented yet so I just did stuff. What exactly do you want the behavior to be of the api? Should you be able to start it playing from the api or only add new songs to the queue?

Reply:
you can pass `member=True` to the `authenticated()` decorator or you could check once you have the guild object in shard.

As far as what to do, it should be the same for using the command: if the user is not in a vc, error. if the user is in a different vc, error. if the user is in a vc and the bot is not, join the user's voice.